docs(docsite): add comparison table for rendering strategies (#1055)
microsoft/fluentui-system-icons: packages/docsite/stories/Icons/UserGuidance.mdx

<h1 className="sbdocs-title">User Guidance</h1>

`@fluentui/react-icons` supports three rendering strategies: **inline SVG** (default), **icon fonts**, and **SVG sprites** (⚠️ Alpha). Each has different trade-offs for performance, caching, and styling flexibility.

> **Tip:** Regardless of which rendering approach you choose, always use **atomic imports** for best tree-shaking and bundle size. See the [Atomic API](../?path=/docs/icons-atomic-api--docs) page for details.

## Choosing the right approach

> <table>

Reviewer: nit: can we use markdown for table ?

Author: fixed

  <thead>
    <tr>
      <th></th>
      <th>Inline SVG (default)</th>
      <th>Icon Font</th>
      <th>SVG Sprites (⚠️ Alpha)</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        <strong>Best for</strong>
      </td>

>       <td>General purpose, most apps</td>

Reviewer: ```suggestion
      <td>General purpose, most apps with very few icons</td>
```

      <td>Large apps with many icons on screen</td>

>       <td>Large or dynamic icon sets served via CDN; avoiding JS bundle overhead</td>

Reviewer: not sure about the "served via CDN" , what's the message there ?

Author: It was about need to load a lot of SVG icons when sprites are needed. For which situations it's best for

Author: I removed second part as it's not needed

        <strong>Styling</strong>
      </td>
      <td>
        Full CSS + <code>primaryFill</code> prop
      </td>
      <td>CSS only (color, font-size)</td>
      <td>
        Full CSS + <code>primaryFill</code> prop
      </td>
    </tr>
    <tr>
      <td>
        <strong>Tree-shaking</strong>
      </td>
      <td>No static asset to subset</td>
      <td>Font file subsetted to used glyphs only</td>
      <td>Unused sprites are never fetched</td>
    </tr>
    <tr>
      <td>
        <strong>Visual fidelity</strong>
      </td>
      <td>Pixel-perfect at all sizes</td>
      <td>Slight reduction at small sizes</td>
      <td>Pixel-perfect at all sizes</td>
    </tr>
    <tr>
      <td>
        <strong>Accessibility</strong>
      </td>
      <td>Full — native SVG semantics</td>
      <td>Full — implementation covers a11y range (see caveats below)</td>
      <td>Full — native SVG semantics</td>
    </tr>
    <tr>
      <td>
        <strong>Loading</strong>
      </td>
      <td>Bundled with JS — no flash</td>
      <td>FOIT / FOUT risk during font load</td>
      <td>One request per sprite group; webpack plugin can merge into a single request</td>
    </tr>
    <tr>
      <td>

>         <strong>Render performance</strong>

Reviewer: lets move "rendering" rows close to each other

Author: fixed

      </td>
      <td>Worst — one SVG DOM subtree per icon</td>
      <td>Best — single glyph, minimal DOM</td>
      <td>
        Good — single DOM node via <code>&lt;use&gt;</code>
      </td>
    </tr>
    <tr>
      <td>
        <strong>Setup</strong>
      </td>
      <td>None (default)</td>
      <td>Webpack config + subsetting plugin</td>

>       <td>Sprite hosting + same-origin proxy + subsetting plugin</td>

Reviewer: ```suggestion
      <td>Webpack subsetting plugin, static assets hosting (font)</td>
      <td>Webpack subsetting plugin, static assets hosting (sprite), same-origin proxy for CDN</td>
```

    </tr>
  </tbody>
</table>

> > **Note:** A **base** rendering mode (vanilla CSS, no Griffel runtime) is available as an alpha opt-in via `/base/*` imports (e.g. `@fluentui/react-icons/base/svg/access-time`). With CSS extraction it reduces JS bundle sizes significantly — up to −70% for font icons, −56% for SVG sprites, −14% for inline SVG. This table will be updated once the base API ships in stable.

Reviewer: back at headless 

```suggestion
> **Note:** A **headless** rendering mode (vanilla CSS, no Griffel runtime) is available as an alpha opt-in via `/headless/*` imports (e.g. `@fluentui/react-icons/headless/svg/access-time`). With CSS extraction it reduces JS bundle sizes significantly — up to −70% for font icons, −56% for SVG sprites, −14% for inline SVG. This table will be updated once the base API ships in stable.
```

### When to use inline SVG (default)

> This is the recommended approach for most applications:

Reviewer: we shouldn't recommend this as it's basically worst solution whilst "works everywhere without your input (setting plugins, reverse-proxy for cdn etc" )
```suggestion
This is the default approach for icons api and used in most applications:
```

You can also enable font icons without the subsetting plugin by only adding the `resolve.conditionNames` configuration. However, this is **not recommended** as it will include the full font files for all icon variants.

### When to use SVG sprites (⚠️ Alpha)

> **Alpha prerelease:** SVG sprites support is available as an alpha prerelease only. Install via `npm i @fluentui/react-icons@alpha`. Breaking changes may occur before general availability.

> Consider SVG sprites when:

Reviewer: I think this should be something like:
- you use lot of icons, you want best fidelity, full styling support, faster rendering than inline svg and smaller js bundles

Author: fixed

> - **Runtime icon loading**: Icons are fetched on demand and cached by the browser, rather than bundled into your JS

Reviewer: I'd also add some sidenote like : we recommend using our webpack plugin which merges all used application icons into 1 sprite which can be loaded by 1 request or inlined within your html to save additional network request.

Author: fixed
